src/backend/joanie/core/models/products.py

```diff
@@ -246,7 +246,7 @@ def generate_invoice(self):
         order = {
             "invoice_id": reference,
             "customer": {
-                "name": self.owner.get_full_name(),
+                "name": self.owner.addresses.get(main=True).fullname,
```

Comment:
Are we sure that there is only one main==True? I think there should be a db constraint for this.

Reply:
We cannot use an unique constraint here as there must be only one main address per user. But `save` method of the Account model is in charge of checking that there is only one main address per user.

Furthermore, I opened an issue (#44) to not use main address to generate invoice. This point must be addressed at the same time as payment feature.

Reply:
Maybe it's possible with complex db constraints. What about something like this?
```python
from django.db.models.constraints import UniqueConstraint
UniqueConstraint(fields=["owner"], condition=Q(main=True), name='unique_main_address_per_user')
```

Reply:
Thanks, I added this constraint.